### PR TITLE
Tegra minimal init fix boot scripts

### PR DESCRIPTION
**Pull request by liuming50**

### CODE REVIEW

**Comment by madisongh** [comment]
Thanks @liuming50 , looks like some good cleanup here.  I'll take a closer look at the changes when I have a bit more time.

File: recipes-core/initrdscripts/tegra-minimal-init_1.0.bb [hunk 1]
```diff
@@ -19,6 +19,7 @@ do_install() {
     mknod -m 622 ${D}/dev/console c 5 1
     install -d ${D}${sysconfdir}
     install -m 0644 ${WORKDIR}/platform-preboot.sh ${D}${sysconfdir}/platform-preboot
+    sed -i -e "s#@@TNSPEC_BOOTDEV@@#${TNSPEC_BOOTDEV}#g" ${D}${sysconfdir}/platform-preboot
```
Comment (madisongh) [inline recipes-core/initrdscripts/tegra-minimal-init_1.0.bb:22]: With this change, the recipe is now machine-specific, so `PACKAGE_ARCH = "${MACHINE_ARCH}"` needs to be added at the end of the recipe.

File: recipes-core/initrdscripts/tegra-minimal-init/platform-preboot.sh [hunk 2]
```diff
@@ -35,8 +35,22 @@ fi
 if [ -z "$rootdev" ]; then
     echo "[FAIL]"
     if [ -n "$mayberoot" ]; then
-	rootdev="$mayberoot"
+        if [ "`echo $mayberoot | cut -c1-5`" = "UUID=" ]; then
```
Comment (madisongh) [inline recipes-core/initrdscripts/tegra-minimal-init/platform-preboot.sh:39]: I'm fine with trying to support these additional ways to specify the rootfs device, but will the `/dev/disk/by-xxxx` symlinks be present?  I thought it was udev (or systemd-udevd) that created those, and that wouldn't be running in a minimal initramfs.

Using `lsblk` to look up partitions by the specified attributes might work better, maybe?

**Review by madisongh — CHANGES_REQUESTED** [review]

File: recipes-core/initrdscripts/tegra-minimal-init/platform-preboot.sh (hunk 2 above)
Comment (liuming50) [reply recipes-core/initrdscripts/tegra-minimal-init/platform-preboot.sh:39]: @madisongh thanks for the input, you are absolutely right, will change that in V2.

**Comment by liuming50** [comment]
Changes in patch set V2:

1 Add "PACKAGE_ARCH = "${MACHINE_ARCH}"" to tegra-minimal-init_1.0.bb as suggested by @madisongh 

2 Drop the udev rules dependencies, change to use 'blkid' as suggested by @madisongh 

3 Add one more fix to rootwait condition checking in init-boot.sh

**Review by madisongh — APPROVED** [review]
Updated changes LGTM.  Thanks again!

**Pull request merged**